src/main/java/pl/sdacademy/carrental/services/CarService.java

```diff
@@ -26,4 +27,12 @@ public CarService(final CarRepository carRepo) {
    public List<Car> getAvailable() {
       return carRepo.findCarsByCurrentStatusEquals(Status.IN);
    }
+
+   public long countAvailableCars(){
+      return getAvailable().size();
+   }
+
+   public boolean availableOnlyOneOrTwoCars(){
+      return countAvailableCars() == 1 || countAvailableCars() == 2;
```

Comment:
ratuje Cię 1st level cache przez wykonywaniem zapytania wielokrotnie :).
Można zmienić na countAvailableCars() < 3 i ta 3 powinna mieć nazwę (magic number in da code)

Reply:
Co do tego to miałem wątpliwości, czy mogę tak zrobić, ponieważ informacja chyba miała zawiadamiać, że samochody jeszcze są ale tylko 1 lub 2. A " < 3 " zawierałoby też przypadek zerowy

Reply:
ok, to wtedy count między 0 a 3. Tak czy inaczej nie wywołuj metody `countAvailableCars` 2 razy (co jeśli trwa ona długo i nie jest cachowana? Akurat w tym przypadku hibernate Cię rature)